frontend/server/k8s-helper.ts

```diff
@@ -80,11 +80,11 @@ function getNameOfViewerResource(logdir: string): string {
  * Create Tensorboard instance via CRD with the given logdir if there is no
  * existing Tensorboard instance.
  */
-export async function newTensorboardInstance(logdir: string, podTemplateSpec: Object = defaultPodTemplateSpec): Promise<void> {
+export async function newTensorboardInstance(logdir: string, tfversion: string, podTemplateSpec: Object = defaultPodTemplateSpec): Promise<void> {
   if (!k8sV1CustomObjectClient) {
     throw new Error('Cannot access kubernetes Custom Object API');
   }
-  const currentPod = await getTensorboardInstance(logdir);
+  const currentPod = await getTensorboardInstance(logdir, tfversion);
```

Comment:
What if there's a current pod with a different version?
Your steps below would fail because you are creating another viewer with the same name.

Reply:
After some discussion with Jing, we find we are facing two choices:
1. Allow user to run more than one version at the same time, so with need to distinguish pods name by version. To implement this way, there might be several tensorboard deployments running at background, and the frontend will display which one will be opened.
2. Another choice is only allow one version of tensorboard running at the same time. Because users mainly hope to choose a version that can be used correctly, this implementation might be functionally enough while keep the page concise. To implement this way, the "logdir" would be an unique identifier of the tensorboard folder. It's not until the user delete the previous version that he can open a new version, within the same logdir. 

Maybe we can discuss later to decide which one to choose!

Reply:
2 sgtm
Do you have a preference?